docs/how-to/reactive.md

```diff
@@ -0,0 +1,112 @@
+
+Note, that the specific revisions/channels in the steps are only marked here to have the howto reproducible, you should adapt these to your needs.
+
+We assume that you have a machine model (named "machine-model") for the runners and MongoDB, and a k8s model (named "k8s-model") for the webhook router.
+
+### GitHub Runner flavors
+
+For this howto, we decided to have deployed three runner flavors: `large`, `large-arm`, `small` . We need
+to deploy those with these names, to comply with the routing table defined below.
+
+```shell
+juju switch machine-model
+juju deploy github-runner large --channel latest/stable ....
+juju deploy github-runner large-arm --channel latest/stable ....
+juju deploy github-runner small --channel latest/stable ....
+```
+
+More steps are required if you are deploying runners on OpenStack cloud.
+
+Please refer to [How to spawn OpenStack runner](how-to/openstack-runner.md) or [How to run on LXD cloud](how-to/run-on-lxd.md)
+for more information on how to deploy the runners.
+
+### MongoDB
+
+You need to deploy a MongoDB application to use as a message queue. 
+You can choose to use the machine charm or the k8s charm version, although we recommend using
+the machine charm as the k8s version may not be reachable from outside the k8s cluster.
+
+```shell
+juju switch machine-model
+juju deploy mongodb --channel 6/edge --revision 188 
+juju expose mongodb
+juju offer mongodb:database
+juju grant <user-in-k8s-cloud> consume <offer-name>
+```
+
+Integrate with the runner charms
+
+```shell
+juju integrate large mongodb
+juju integrate large-arm mongodb
+juju integrate small mongodb
+```
+
+### Define a webhook in your organisation or repository where the self-hosted runners are registered.
+
+On your repository or organisations page on Github, you need to go to the settings and create a Webhook
+(e.g. https://github.com/canonical/github-runner-operator/settings/hooks). Please make sure to select
+
+- the Webhook url to be the URL of the webhook router
+- the content type `application/json`
+- the secret you defined in the webhook router (if you have so, which is recommended for security reasons)
+- the individual event "Workflow jobs" (and only this, as all other events will just be rejected by the webhook router)
+
+### Webhook router
+
+The webhook router is a k8s charm, therefore you need to deploy it on a k8s model.
+
+First, define a routing table to decide which labels should be routed to which runner flavor
+
+```shell
+cat <<EOF > routing_table.yaml 
+- large: [large, x64]
+- large-arm: [large, arm64]
+- small: [small]
+EOF
+```
+
+We decide to route all jobs with any label combination in the set `large,x64` to the large flavor, `large,arm64` to large-arm,
+and labels with `small` to small.
+This means, depending on which labels your users are setting in the workflow file, a different runner flavor will be used to
+execute the job.
+
+
```

Comment:
I think it would be nice to have a preamble here to introduce the code block. Something like, "Switch to the K8s model and deploy the webhook router charm:"